src/EditModel/event-program/assign-data-elements/epics.js

```diff
@@ -118,6 +121,32 @@ const removeDataElementFromStage = store => action$ =>
             programStageToEdit.programStageDataElements = removeDataElements(
                 programStageDataElements
             );
+
+            // Remove DataElements from any sections that contains the removed DataElements
+            let programStageSections = getStageSectionsById(
+                state,
+                programStageToEditId
+            );
+
+            if (programStageSections && Object.keys(programStageSections).length > 0) {
+                state.programStageSections[
+                    programStageToEditId
+                ] = programStageSections.map(section => {
+
+                    const dataElementsRemoved = section.dataElements.filter(
+                        element => !dataElementIdsToRemove.includes(element.id)
+                    );
```

Comment:
The logic in here is completely sound, but I think I would have been able to comprehend things a bit more easily if the const `dataElementsRemoved` would have been named differently. Perhaps `cleanedDataElements`?

I realise the suggestion above is pretty subjective, so feel free to just merge this in as-is.

Reply:
Thanks, I actually struggled with the name here as I was quite exhausted when I did this. I think I will go with `updatedDataElements` as that seems to be used elsewhere.